packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -0,0 +1,257 @@
+# stdlib
+import functools
+import itertools
+import operator
+import secrets
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+# third party
+import numpy as np
+
+# syft absolute
+from syft.core.tensor.passthrough import PassthroughTensor
+from syft.core.tensor.smpc.share_tensor import ShareTensor
+
+# relative
+from .utils import ispointer
+
+METHODS_FORWARD_ALL_SHARES = {
+    "t",
+    "squeeze",
+    "unsqueeze",
+    "view",
+    "sum",
+    "clone",
+    "flatten",
+    "reshape",
+    "repeat",
+    "narrow",
+    "dim",
+    "transpose",
+}
+
+
+class MPCTensor(PassthroughTensor):
+    def __init__(
```

Comment:
Should we have  `slots`  or we will keep it for a later PR?

Reply:
I think we can add ```__slots__``` later or maybe not at all. ```__slots__``` can have a disadvantage when used in an inheritance scenario and in this case we (the SMPC team) do not fully control what happens with the ```PassthroughTensor```)